### PR TITLE
Update developer docs to represent the current package layout

### PR DESCRIPTION
**Pull request by wagoodman**
This updates the developer docs with the latest patterns (this will be changed even further in an upcoming PR).

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -149,7 +127,10 @@ sequenceDiagram
     Note right of catalog: cataloger configuration is done based on src
 ```
 
-### Package object
+Additionally, here is a [gist of using syft as a library](https://gist.github.com/wagoodman/57ed59a6d57600c23913071b8470175b) to generate a SBOM for a docker image.
```
Comment (spiffcs) [inline DEVELOPING.md:130]: It looks like a user has already left a comment on this gist mentioning it's out of date - let me try it with the latest syft library

Comment (spiffcs) [reply DEVELOPING.md:130]: Yea running into the same issue they hi-lighted:
```
./main.go:16:30: undefined: source.New
./main.go:33:77: cannot use source.SquashedScope (constant "Squashed" of type source.Scope) as cataloger.Config value in argument to syft.CatalogPackages
./main.go:38:19: result.Artifacts.PackageCatalog undefined (type "github.com/anchore/syft/syft/sbom".Artifacts has no field or method PackageCatalog)
./main.go:39:19: result.Artifacts.Distro undefined (type "github.com/anchore/syft/syft/sbom".Artifacts has no field or method Distro)
./main.go:43:21: undefined: syft.Encode
./main.go:43:43: undefined: format.JSONOption
```

Comment (spiffcs) [reply DEVELOPING.md:130]: This is the gist I had put together around the same time using the detection portion of the source library:
https://gist.github.com/spiffcs/3027638b7ba904d07e482a712bc00d3d

File: DEVELOPING.md [hunk 2]
```diff
@@ -61,35 +61,30 @@ For more information on this setup and troubleshooting see [issue 1895](https://
...
+│   ├── cli/
+│   │   ├── cli.go          // where all commands are wired up
+│   │   ├── commands/       // all command implementations
+│   │   ├── options/        // all command flags and configuration options
+│   │   └── ui/             // all handlers for events that are shown on the UI
+│   └── main.go             // entrypoint for the application
+└── syft/                   // the "core" syft library
+    ├── format/             // contains code to encode or decode to and from SBOM formats
+    ├── pkg/                // contains code to catalog packages from a source
+    ├── sbom/               // contains the definition of an SB OM
```
[the remark this answers is not in the document]
Comment (wagoodman) [reply DEVELOPING.md:76]: nice catch, I'll get that fixed

File: DEVELOPING.md (hunk 1 above)
Comment (wagoodman) [reply DEVELOPING.md:130]: I haven't updated the gist itself, and this is one of the main changes that will be in an upcoming PR. For the meantime I can update to show your gist until the other PR lands.

**Pull request merged**